packages/token/src/erc20/interface.cairo

```diff
@@ -78,35 +169,77 @@ pub trait IERC20Permit<TState> {
         deadline: u64,
         signature: Span<felt252>,
     );
+
+    /// Returns the current nonce for `owner`.
     fn nonces(self: @TState, owner: ContractAddress) -> felt252;
+
+    /// Returns the domain separator used in the encoding of the signatures for permit.
     fn DOMAIN_SEPARATOR(self: @TState) -> felt252;
 }
 
+/// Interface that combines all ERC20 interfaces including permit and metadata.
 #[starknet::interface]
 pub trait ERC20ABI<TState> {
     // IERC20
+    /// Returns the total supply of tokens.
```

Comment:
Similar to mixins, we don't need individual function comments for ABIs, just the grouping